src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1158,3 +1163,11 @@ int sd_dhcp_server_set_ntp(sd_dhcp_server *server, const struct in_addr ntp[], u
 
         return 1;
 }
+
+int sd_dhcp_server_set_emit_router(sd_dhcp_server *server, int enabled) {
+        assert_return(server, -EINVAL);
+
+        server->emit_router = enabled;
```

Comment:
Other functions here return 0 if the setting is unchanged, and 1 iff it is changed. Please add a check like in sd_dhcp_server_set_default_lease_time().